dipper/sources/ZFIN.py

```diff
@@ -2494,46 +2324,58 @@ def process_fish_disease_models(self, limit=None):
 
         return
 
-    def _map_sextuple_to_phenotype(
-            self, superterm1_id, subterm1_id, quality_id, superterm2_id,
-            subterm2_id, modifier):
+    def _map_octuple_to_phenotype(
+             self,
+             subterm1_id,
+             post_composed_relationship_id_1,
+             superterm1_id,
+             quality_id,
+             subterm2_id,
+             post_composed_relationship_id_2,
+             superterm2_id,
+             modifier):
         """
-        This will take the 6-part EQ-style annotation
-        used by ZFIN and return the ZP id.
-        Currently relies on an external mapping file,
-        but the method may be swapped out in the future
-        :param superterm1_id:
-        :param subterm1_id:
-        :param quality_id:
-        :param superterm2_id:
-        :param subterm2_id:
-        :param modifier:
-        :return: ZP id
+        This will take the 8-part EQ-style annotation
+        used in zp-mapping.txt and return the ZP id.
+        
+        :param subterm1_id,
+        :param post_composed_relationship_id_1,
+        :param superterm1_id,
+        :param quality_id,
+        :param subterm2_id,
+        :param post_composed_relationship_id_2,
+        :param superterm2_id,
+        :param modifier
         """
+
         zp_id = None
 
         # zfin uses free-text modifiers,
         # but we need to convert them to proper PATO classes for the mapping
         mod_id = self.resolve(modifier, False)
```

Comment:
I am not sure why the actual PATO modifier is needed at all here. Why does not the string 'abnormal' suffice? I am only asking because I would like to be sure that mod_id is not used anywhere for anything else other than resolving the mapping.. (especially not added to the knowledge graph).

Reply:
generally the way things work in Dipper is to look things like this ("abnormal" in this case) in the translation table using .resolve, and match things using ontology ids where possible. this prevents things like "abnormal" not matching "abnormal " or "Abnormal" or " abnormal ", among other advantages.